### PR TITLE
refactor(levm): memory module

### PR DESCRIPTION
**Pull request by ilitteri**
**Motivation**

- The `Memory` implementation was similar to type aliasing a `Vec<u8>`. The current implementation unnecessarily adds an extra layer of indirection.
- The current `Memory` implementation has some bugs and not all its methods are panic safe.

**Description**

- `Memory` is now a type alias for `Vec<u8>`.
- All the `Memory` "methods" are implemented as functions in the `memory` module and called taking advantage of the namespacing (e.g. `memory::load_word(...)`).
- All the `Memory` functions are panic safe.
- Memory access and expansion cost functions were refactored and moved to the `memory` module. As a side note, in a future refactor we will be updating the gas cost type from `U256` to `usize`, this PR kind of introduces this notion, but it turns the `usize`s into `U256` wherever is necessary for backward compatibility the implementation.

### CODE REVIEW

**Comment by ilitteri** [comment]
I think the problem with the EF Tests run will be fixed after https://github.com/lambdaclass/ethrex/pull/1394 is merged.

**Review by JereSalo — APPROVED** [review]
It looks good overall. I didn't find any issues in the code.

File: crates/vm/levm/src/opcode_handlers/system.rs [hunk 1]
```diff
@@ -195,7 +195,7 @@ impl VM {
 
         self.increase_consumed_gas(
             current_call_frame,
-            memory::expansion_cost(offset, size)?.into(),
+            memory::expansion_cost(offset + size, current_call_frame.memory.len())?.into(),
```
Comment (JereSalo) [inline crates/vm/levm/src/opcode_handlers/system.rs]: should be checked_add

**Review by JereSalo — APPROVED** [review]
LGTM

**Pull request merged**